vcf_stuff/filtering/annotate_somatic_vcf.smk

```diff
@@ -150,71 +155,11 @@ rule somatic_vcf_regions_anno:
         'vcfanno {input.toml} {input.vcf} | bgzip -c > {output.vcf} && '
         'tabix -f -p vcf {output.vcf}'
 
-
-# Possibly subset VCF to avoid PCGR choking with R stuff.
-# too higly mutated samples might indicate germline contamination.
-# If the noise wasn't germline, it might be artefacts/errors from FFPE or ortherwise low quality data.
-# subsetting to cancer genes in this case.
-rule maybe_subset_highly_mutated:
-    input:
-        vcf = rules.somatic_vcf_regions_anno.output.vcf,
-        tbi = rules.somatic_vcf_regions_anno.output.tbi,
-    output:
-        vcf = f'somatic_anno/subset/{SAMPLE}-somatic.vcf.gz',
-        tbi = f'somatic_anno/subset/{SAMPLE}-somatic.vcf.gz.tbi',
-        subset_highly_mutated_stats = f'somatic_anno/subset_highly_mutated_stats.yaml',
-    params:
-        no_gnomad_vcf = f'somatic_anno/subset/no_gnomad/{SAMPLE}.vcf.gz',
-    run:
-        total_vars = count_vars(input.vcf)
-        vars_no_gnomad = None
-        vars_cancer_genes = None
-        if total_vars > 500_000:
-            warn(f'Found {total_vars}>500k somatic variants, start with removing gnomAD_AF>0.01')
```

Comment:
I feel this would still be a useful filter regardless for highly mutated or somewhat contaminated samples. Not sure it is needed for this current fix but I'd like to re-introduce this in the future (or maybe have it as a generic filter given that the DRAGEN PoN filter seems to be less effective than the current one).